kernel/sched.c

```diff
@@ -1338,45 +1338,64 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq)
 }
 
 #ifdef CONFIG_SCHED_MULTIQ
-# if (K_LOWEST_THREAD_PRIO - K_HIGHEST_THREAD_PRIO) > 31
-# error Too many priorities for multiqueue scheduler (max 32)
-# endif
+
+struct prio_info {
+	uint8_t offset_prio;
+	uint8_t idx;
+	uint8_t bit;
+};
+
+static inline struct prio_info get_prio_info(int8_t old_prio)
+{
+	struct prio_info ret;
+
+	ret.offset_prio = old_prio - K_HIGHEST_THREAD_PRIO;
+	ret.idx = ret.offset_prio / NBITS;
+	ret.bit = ret.offset_prio % NBITS;
```

Comment:
Mod is an expensive operation on some architectures. Can the same thing be achieved a different way?

Do we already have something in e.g. `sys/util.h` do get the index and bit of a multi-word bitmap entry?

Reply:
> Mod is an expensive operation on some architectures. Can the same thing be achieved a different way?
> 
yes, there is a tradeoff.
We can enforce bitmap array is 2^x and the mod operation will be replace with &, but it will need additional memory.
This is the reason my suggestion in doc as below.
‘’’
This scheduler will invoke mod operation so it is suggested
	  to set CONFIG_NUM_PREEMPT_PRIORITIES + CONFIG_NUM_COOP_PRIORITIES
	  to power of 2 minus 1 (e.g. 31, 63, 127...).
‘’’

> Do we already have something in e.g. `sys/util.h` do get the index and bit of a multi-word bitmap entry?

I didn’t see that.